tasks/bootstrap_local_vm/01_prepare_routing_rules.yml

```diff
@@ -39,20 +51,34 @@
   - name: Refresh network facts
     setup:
     tags: [ 'skip_ansible_lint' ]
-  - name: Prepare CIDR for {{ virbr_default }}
+  - name: Fetch IPv4 CIDR for {{ virbr_default }}
     set_fact:
-      virbr_cidr: "{{ (hostvars[inventory_hostname]['ansible_'+virbr_default]['ipv4']['address']+'/'+hostvars[inventory_hostname]['ansible_'+virbr_default]['ipv4']['netmask']) |ipv4('host/prefix') }}"
-  - name: Add outbound route rules
-    command: ip rule add from {{ virbr_cidr }} priority 101 table main
+      virbr_cidr_ipv4: "{{ (hostvars[inventory_hostname]['ansible_'+virbr_default]['ipv4']['address']+'/'+hostvars[inventory_hostname]['ansible_'+virbr_default]['ipv4']['netmask']) |ipv4('host/prefix') }}"
+  - name: Add IPv4 outbound route rules
+    command: ip rule add from {{ virbr_cidr_ipv4 }} priority 101 table main
+    environment: "{{ he_cmd_lang }}"
+    register: result
+    when: not ipv6_deployment and "\"101:\tfrom \"+virbr_cidr_ipv4+\" lookup main\" not in route_rules_ipv4.stdout"
```

Comment:
This `when` statement is used 4 times in this playbook and it's quite ugly. Maybe it is better to save it as a `fact` and then reuse it for better readability

Reply:
Unfortunately they are similar but not really the same